lab-michael/routes/recipe-routes.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const storage = require('../lib/storage.js');
+const response = require('../lib/response.js');
+const Recipe = require('../model/recipe.js');
+
+module.exports = function(router) {
+  router.get('/api/recipe', function(req, res){
+    if(req.url.query.id){
+      storage.fetchItem('recipe', req.url.query.id)
+      .then(recipe => {
+        res.writeHead(200, {
+          'Content-Type': 'application/json',
+        });
+        res.write(JSON.stringify(recipe));
+        res.end();
+      })
+      .catch(err => {
+        console.error(err);
+        res.writeHead(404, {
+          'Content-Type': 'text/plain',
+        });
+        res.write('not found\n');
+        res.end();
+      });
+      return;
+    } else {
+      if(!req.url.query.id){
+        storage.fetchAll('recipe', req.url.query.id)
+        .then(recipe => {
+          res.writeHead(200, {
+            'Content-Type': 'application/json',
+          });
+          res.write(JSON.stringify(recipe));
```

Comment:
I believe you're just writing out the file names at this point, rather than receiving each file's content and displaying All Recipes.